libs/librepcb/projecteditor/boardeditor/fsm/bes_addvia.cpp

```diff
@@ -283,45 +292,65 @@ BES_Base::ProcRetVal BES_AddVia::processSceneEvent(BEE_Base* event) noexcept {
 }
 
 bool BES_AddVia::addVia(Board& board) noexcept {
-  Q_ASSERT(mUndoCmdActive == false);
-  Q_ASSERT(mCurrentViaNetSignal);
+  Q_ASSERT(mSubState == SubState_Idle);
+  Q_ASSERT(!mUndoStack.isCommandGroupActive());
 
   try {
+    mSubState = SubState_PositioningVia;
     mUndoStack.beginCmdGroup(tr("Add via to board"));
```

Comment:
The state must be set *after* starting the new undo command, since starting a command raises an exception if there is already a command active. Then we don't want to be in `SubState_PositioningVia` state since this would make `abortPlacement()` aborting a command which was started by another tool.

```suggestion
    mUndoStack.beginCmdGroup(tr("Add via to board"));
    mSubState = SubState_PositioningVia;
```

Reply:
Will fix it there too